intelmq/bin/intelmqctl.py

```diff
@@ -945,13 +945,14 @@ def check(self, no_connections=False):
                 if bot_check:
                     for log_line in bot_check:
                         getattr(check_logger, log_line[0])(f"Bot {bot_id!r}: {log_line[1]}")
-        for group in utils.list_all_bots().values():
-            for bot_id, bot in group.items():
-                if subprocess.call(['which', bot['module']], stdout=subprocess.DEVNULL,
-                                   stderr=subprocess.DEVNULL):
-                    check_logger.error('Incomplete installation: Executable %r for %r not found in $PATH (%r).',
-                                       bot['module'], bot_id, os.getenv('PATH'))
-                    retval = 1
+        if check_executables:
+            for group in utils.list_all_bots().values():
+                for bot_id, bot in group.items():
+                    if subprocess.call(['which', bot['module']], stdout=subprocess.DEVNULL,
```

Comment:
assumes that `which` is in the $PATH. Might want to make this explicit. Also think about docker images where the `which` command might be stripped out ? Would there be a more self-reliant mechanism to check if the executable exists? Like we should know, where it was installed, no?

Reply:
Addendum: if you think it's ok enough in all setups where intelmqctl could be called , to use the `which` command, then I am fine with this line. I can't assess it myself.

Reply:
ignore my comment... sorry.

Reply:
This PR only adds the enclosing `if check_executables:` condition, not the subprocess call, which is in place since 2017 (05505037866ee84cb214263a88d6adef61c2d4de)